### PR TITLE
Airport Challenge

### PR DESCRIPTION
**Pull request by RoystonHenson**

### CODE REVIEW

File: lib/airport.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  def initialize(capacity=DEFAULT_CAPACITY)
+    @planes = []
+    @capacity = capacity
+  end
+
+  def land(plane)
+    weather_good_today?
+    raise "That plane has already landed!" unless !@planes.include?(plane)
+    raise "You cannot land in a storm" unless weather_today_good?
```
Comment (maryrosecook) [inline lib/airport.rb]: I'm not sure this method exists?

File: lib/airport.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    raise "This airport is full!" unless !full?
+    confirm_landing
+    @planes << plane
+  end
+
+  def take_off(plane)
+    weather_good_today?
+    raise "That plane isn't in this airport!" unless @planes.include?(plane)
+    raise "You cannot take off in a storm!" unless weather_today_good?
+    confirm_take_off
```
Comment (maryrosecook) [inline lib/airport.rb]: This logging, though helpful for debugging, should probably be removed in the final version.

File: lib/airport.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  def confirm_take_off
+    puts "The plane has taken-off!"
+  end
+
+  def full?
+    return true if @planes.length >= @capacity
+  end
+
+  def weather_good_today?
```
Comment (maryrosecook) [inline lib/airport.rb]: Perhaps move weather to its own class?

File: lib/airport.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    @weather = (rand(11))
+    if @weather < 6
```
Comment (maryrosecook) [inline lib/airport.rb]: Can this if be condensed to a single line?

File: lib/airport.rb [hunk 5]
```diff
@@ -43,10 +42,6 @@ def full?
 
   def weather_good_today?
     @weather = (rand(11))
-    if @weather < 6
-      true
-    else
-      false
-    end
+    true unless @weather > 8
```
Comment (maryrosecook) [inline lib/airport.rb]: Perhaps even simply `@weather <= 8`

**Comment by RoystonHenson** [comment]
Thanks Mary :)


Last weekend I had this method setup to call random weather every time I tried to land/take-off a plane(thinking that weather can change throughout the day). But I forgot to commit it to github before experimenting, broke it and ran out of time to fix it and get the tests passing before having to submit the pull request.

I added it back as a method last night and was wondering if that method would be fine or is it still better to make weather a class?


best

Royston



________________________________

**Comment by maryrosecook** [comment]
Having weather as a class reduces the number of responsibilities of the Airport, which is a good thing.  But having the weather in the Airport isn't the end of the world!

**Comment by RoystonHenson** [comment]
Ok. Thanks again Mary!


best

Royston


________________________________

**Pull request closed without merging**